### ISSUE TITLE
Consider combining pep8 + pyflakes check into just flake8

### ISSUE THREAD

**Issue by Ivoz** [gevent/gevent#727]
What it says on the tin!

- https://flake8.readthedocs.org/en/latest/
- [configuring](https://flake8.readthedocs.org/en/latest/config.html#per-project)

**Comment by jamadden** [comment]
I'm personally content with the current setup. There's some custom tooling in place to run `pyflakes` that would need to be looked it if a change is to be made.

**Comment by jamadden** [comment]
(*If* we were to talk about changing from pyflakes to something else, my personal preferred linter happens to be [pylint](http://www.pylint.org).)

**Comment by thedrow** [comment]
Or just landscape.io? Would that cover our needs?

**Comment by jamadden** [comment]
It has to be something that not only operates as something that can automatically fail the builds of PRs that don't meet the standards (can landscape.io do that?) but which can be operated locally by authors and contributors to perform the same checks before committing/submitting PRs. Existing solutions include:


|          Purpose         |                   Local                                             |     Remote     |
|------------------------|---------------------------------------------------------------|--------------|
| Multiple Python Versions               | tox                                              | Travis         |
| Lint                                                 | pep8/pyflakes                             | TASK=lint-py27 |
| Coverage                                       | testrunner --coverage/coverage html | coveralls.io |
| Randomized Masochistic Torture    | Cover self with honey, poke sleeping bear | Appveyor       |

**Comment by jamadden** [comment]
hehe, it looks like landscape runs pylint behind the scenes :)

**Comment by jamadden** [comment]
Interesting. landscape uses [prospector](https://github.com/landscapeio/prospector) to do its checks. prospector is a wrapper around pylint, pep8, pyflakes, pyroma, and a few other tools. That could possibly be a comprehensive replacement for our current setup, with the same configuration work mentioned above.

**Comment by thedrow** [comment]
Yes, landscape can fail builds and prospector can be used locally as well.
Also, I just LOLed.

**Comment by jamadden** [comment]
I'm working in the [landscape branch](https://github.com/gevent/gevent/compare/landscape) to test this out, and I have enabled the [landscape reports](https://landscape.io/github/gevent/gevent). So far, I'm liking prospector and the way it can be configured (much better than our ad-hoc regexes in `util/pyflakes.py`). The only question will be if it's flexible enough to get us down to 0 errors, and how much work (or compromise of ignoring errors) that will take.

**Comment by jamadden** [comment]
I've merged #769, which switches to the much more configurable prospector. Under the covers it still uses pep8, pyflakes and adds pylint, pyroma, and dodgy. As noted above, I've enabled lanscape reports (hmm, should see if they have a badge), and I have landscape set to fail PRs if errors are introduced (we're still running the lint tests on Travis too, because we need to run them on both Python 2 and Python 3; lanscape covers 3, travis covers 2, and travis also fails). 

prospector/pylint actually happened to uncover a couple of genuine bugs which were fixed in the merge.

Thanks @thedrow
